tools/protovalidate-conformance/internal/cases/cases_message.go

```diff
@@ -155,5 +155,36 @@ func messageSuite() suites.Suite {
 				},
 			),
 		},
+		"oneof/single-field/valid": {
+			// BoolField is not part of the oneof, so it can be set on this message
+			Message:  &cases.MessageOneofSingleField{StrField: "test", BoolField: true},
+			Expected: results.Success(true),
+		},
+		"oneof/two-fields/valid": {
+			Message:  &cases.MessageOneofMultipleFields{StrField: "test"},
+			Expected: results.Success(true),
+		},
+		"oneof/two-fields/invalid": {
+			Message: &cases.MessageOneofMultipleFields{StrField: "test", BoolField: true},
+			Expected: results.Violations(
+				&validate.Violation{
+					RuleId:  proto.String("message.oneof"),
+					Message: proto.String("only one of ['str_field', 'bool_field'] can be set"),
+				},
+			),
+		},
+		"oneof/multiple-fields/required/valid": {
+			Message:  &cases.MessageOneofMultipleFieldsRequired{StrField: "test"},
+			Expected: results.Success(true),
+		},
+		"oneof/multiple-fields/required/invalid": {
+			Message: &cases.MessageOneofMultipleFieldsRequired{},
+			Expected: results.Violations(
+				&validate.Violation{
+					RuleId:  proto.String("message.oneof"),
+					Message: proto.String("one of ['str_field', 'bool_field'] must be set"),
+				},
+			),
+		},
```

Comment:
Can we add a compile error for unknown names in `MessageOneofRule.fields`?

Without an error, it's awfully easy to end up with silently broken validation, for example if renaming a field.

```suggestion
		},
		"oneof/unknown_field_name/compile_error": {
			Message:  &cases.MessageOneofUnknownFieldName{},
			Expected: results.CompilationError("field \"xxx\" not found in message buf.validate.conformance.cases.MessageOneofUnknownFieldName"),
		},
```

We'll need this addition to proto/protovalidate-testing/buf/validate/conformance/cases/messages.proto:

```proto

message MessageOneofUnknownFieldName {
  option (buf.validate.message).oneof = {
    fields: ["xxx"]
  };
}
```

I've put up this PR for Go to support it: https://github.com/bufbuild/protovalidate-go/pull/255

https://github.com/bufbuild/protovalidate-es/pull/45 and https://github.com/bufbuild/protovalidate-python/pull/317 support it too. I'm happy to revert if we decide we don't want this compile error, but it seems important to me.